test/framework/flux.go

```diff
@@ -231,9 +231,9 @@ func (e *E2ETest) validateGitopsRepoContent(gitOptions *GitOptions) {
 
 func (e *E2ETest) convertVSphereMachineConfigs(cpName, workerName string, vsphereMachineConfigs map[string]*v1alpha1.VSphereMachineConfig) []providers.MachineConfig {
 	var configs []providers.MachineConfig
-	configs = append(configs, *vsphereMachineConfigs[cpName])
+	configs = append(configs, vsphereMachineConfigs[cpName])
```

Comment:
Are you making this change because you saw an error in dereferencing? Wondering if we want to add a check for nil if that's the case.

Reply:
the values being passed in, `vsphereMachineConfigs`, is a map of string: pointer to machine config. 

So by using a pointer TO THE POINTER, or basically `**vsphereMachineConfig`, we're referencing the concrete type, which does not implement the newly added `OSFamily()` method of the `MachineConfig` interface, as it has a pointer receiver on the `VsphereMachineConfig`

so here we're just using the given pointer from the array, rather than the concrete type `**vsphereMachineConfig`

I'll add a nil check for safey sake, but I think it's pretty safe without it given the current usage and the non-result of appending nil